hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestNSSummaryDiskUsageOrdering.java

```diff
@@ -0,0 +1,421 @@
+
+    ReconTestInjector reconTestInjector =
+        new ReconTestInjector.Builder(temporaryFolder.toFile())
+            .withReconOm(reconOMMetadataManager)
+            .withOmServiceProvider(ozoneManagerServiceProvider)
+            .withReconSqlDb()
+            .withContainerDB()
+            .addBinding(OzoneStorageContainerManager.class,
+                getMockReconSCM())
+            .addBinding(StorageContainerServiceProvider.class,
+                mock(StorageContainerServiceProviderImpl.class))
+            .addBinding(NSSummaryEndpoint.class)
+            .build();
+    ReconNamespaceSummaryManager reconNamespaceSummaryManager =
+        reconTestInjector.getInstance(ReconNamespaceSummaryManager.class);
+    nsSummaryEndpoint = reconTestInjector.getInstance(NSSummaryEndpoint.class);
+
+    // populate OM DB and reprocess into Recon RocksDB
+    populateOMDB();
+    NSSummaryTaskWithFSO nSSummaryTaskWithFso =
+        new NSSummaryTaskWithFSO(reconNamespaceSummaryManager,
+            reconOMMetadataManager, ozoneConfiguration);
+    nSSummaryTaskWithFso.reprocessWithFSO(reconOMMetadataManager);
+  }
+
+  /**
+   * Create a new OM Metadata manager instance with one user, one vol, and two
+   * buckets.
+   * @throws IOException ioEx
+   */
+  private static OMMetadataManager initializeNewOmMetadataManager(
+      File omDbDir)
+      throws IOException {
+    OzoneConfiguration omConfiguration = new OzoneConfiguration();
+    omConfiguration.set(OZONE_OM_DB_DIRS,
+        omDbDir.getAbsolutePath());
+    OMMetadataManager omMetadataManager = new OmMetadataManagerImpl(
+        omConfiguration, null);
+    return omMetadataManager;
+  }
+
+  @Test
+  public void testDiskUsageOrderingForRoot() throws Exception {
+    // root level DU
+    // Verify the ordering of subpaths under the root
+    verifyOrdering(ROOT_PATH);
+  }
+
+  @Test
+  public void testDiskUsageOrderingForVolume() throws Exception {
+    // volume level DU
+    // Verify the ordering of subpaths under the volume
+    verifyOrdering("/volA");
+    verifyOrdering("/volB");
+  }
+
+  @Test
+  public void testDiskUsageOrderingForBucket() throws Exception {
+    // bucket level DU
+    // Verify the ordering of subpaths under the bucket
+    verifyOrdering("/volA/bucketA1");
+    verifyOrdering("/volA/bucketA2");
+    verifyOrdering("/volA/bucketA3");
+    verifyOrdering("/volB/bucketB1");
+  }
+
+  private void verifyOrdering(String path)
+      throws IOException {
+    Response response =
+        nsSummaryEndpoint.getDiskUsage(path, true, false, true);
+    DUResponse duRes = (DUResponse) response.getEntity();
+    List<DUResponse.DiskUsage> duData = duRes.getDuData();
+    List<DUResponse.DiskUsage> sortedDuData = new ArrayList<>(duData);
+    // Sort the DU data by size in descending order to compare with the original.
+    sortedDuData.sort(
+        Comparator.comparingLong(DUResponse.DiskUsage::getSize).reversed());
```

Comment:
Can you pls explain this sorting and verification. As per my understanding, both `duData` and `sortedDuData` will be sorted already, so at this line, are you trying to reverse , then it should be in ascending order and subpaths should not match ? Am I missing something here ?

Reply:
The `verifyOrdering` method is designed to validate that the disk usage data returned from the `nsSummaryEndpoint.getDiskUsage` method is correctly sorted in descending order by size. To check this, it first copies the original data (`duData`) into a new list (sortedDuData), and then sorts this new list in descending order explicitly. By comparing the subpaths of the disk usage objects in both the original and sorted lists, the method confirms that the original data was indeed sorted correctly.

Reply:
but original data 'duData' wouldn't be already sorted ?

Reply:
Right if the original is sorted then the ordering will be the same as sortedDuData and hence this test will pass, but if the original duData is not sorted then the ordering with the sortedDuData will not match hence it will result in failure.
That is where the test fails.

Reply:
Ok thanks @ArafatKhan2198 for explaining in detail. Its ok to have it this way.